salesforce-sync/types.bal

```diff
@@ -0,0 +1,245 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License. 
+import ballerina/time;
+
+# Service supported sync types/objects
+enum SyncObject {
+    ACCOUNT,
+    OPPORTUNITY
+}
+
+# Sync log statuses used on the log table in database
+enum LogStatus {
+    PROCESSING = "Processing",
+    COMPLETED = "Completed",
+    FAILED = "Failed"
+}
+
+// [Config record types]
+# Salesforce sync configuration.
+# 
+# + enabled - Enable/disable sync
+# + periodInHours - Time (minimum) period between sync in hours
+# + soqlQuery - Salesforce SOQL sync query
+type SfSyncConf record {
+    boolean enabled;
+    decimal periodInHours;
+    string soqlQuery;
+};
+
+# Salesforce sync minimal configuration.
+# 
+# + enabled - Enable/disable sync
+# + periodInHours - Time (minimum) period between sync in hours
+type SfSyncMinimalConf record {
+    boolean enabled;
+    decimal periodInHours;
+};
+
+# [Configurable] Salesforce service configuration.
+# 
+# + refreshUrl - OAuth2 refresh token endpoint
+# + refreshToken - OAuth2 refresh token
+# + clientId - OAuth2 client ID
+# + clientSecret - OAuth2 client secret
+type SalesforceAuthConf record {|
+    string refreshUrl;
+    string refreshToken;
+    string clientId;
+    string clientSecret;
+|};
+
+# [Configurable] MySQL Database configuration.
+# 
+# + host - Hostname
+# + user - Username
+# + password - Password
+# + database - Default database
+type DatabaseConf record {|
+    string host;
+    string user;
+    string password;
+    string database;
+|};
+
+# [Util] Salesforce API response error record
+# 
+# + errorCode - Salesforce generated error code
+# + message - Error message
+type SalesforceResponseError record {
+    string errorCode;
+    string message;
+};
+
+# [Database] Sync log time retrieval record.
+# 
+# + id - Database row ID
+# + start_time - Sync start time
+# + end_time - Sync end time
+type DBSyncLogTimes record {|
+    int id;
+    time:Utc start_time;
+    time:Utc end_time;
+|};
+
+# [Salesforce] Account sync query record.
+# 
+# + Id - Account ID
+# + Name - Account name
+# + BillingCountry - Billing country
+# + ShippingCountry - Shipping country
+# + Sales_Regions__c - Sales regions
+# + Sub_Region__c - Sales sub-region
+# + NAICS_Industry__c - NAICS Industry
+# + Sub_Industry__c - NAICS sub-industry
+# + Account_Classification__c - Account classification
+# + ARR_Churn_Date__c - Churn date of the ARR account
+# + Owner - Account owner details
+type SFAccountSyncRecord record {
+    string Id;
+    string Name;
+    string? BillingCountry;
+    string? ShippingCountry;
+    string? Sales_Regions__c;
+    string? Sub_Region__c;
+    string? NAICS_Industry__c;
+    string? Sub_Industry__c;
+    string? Account_Classification__c;
+    string? ARR_Churn_Date__c;
+    record {
+        string Email;
+        string Name;
+    } Owner;
+};
```

Comment:
This the old documentation technique. Please do the new type documentation in-line as follows [1].

[1] https://learn-ballerina.github.io/best_practices/documenting_the_source_code.html#documenting-records